network/src/network_group.rs

```diff
@@ -0,0 +1,49 @@
+use libp2p::core::{AddrComponent, Multiaddr};
+use std::net::IpAddr;
+
+pub type Group = Vec<u8>;
```

Comment:
u8 * 4 -> u32

Since the only usage of Group is to categorize addresses, I believe Vec is more clearly and would never cause a collision between difference sized Vector,

Reply:
This is a waste of %50 space in the most conditions today.